File: 2021/12/09/main.go
package main

func canReach(arr []int, start int) bool {

	if start < 0 || start > len(arr)-1 || arr[start] == -1 {
		return false
	}

	delta := arr[start]
	arr[start] = -1

	return (delta == 0) || canReach(arr, start+delta) || canReach(arr, start-delta)
}

File: 2021/12/09/main.py
class Solution:
    def canReach(self, arr: List[int], start: int) -> bool:
        
        if start < 0 or start > len(arr)-1 or arr[start] == -1:
            return False
        
        delta = arr[start]
        arr[start] = -1

        return  (delta == 0) or self.canReach(arr, start+delta) or self.canReach(arr, start-delta)